pkgs/gitRepo/default.nix

```diff
@@ -0,0 +1,70 @@
+{ lib, inputs, fetchFromGitHub, rsync, git, gnupg, less, openssh, ... }:
+let
+  inherit (inputs) nixpkgs-unstable;
+
+  unstablePkgs = nixpkgs-unstable.legacyPackages.x86_64-linux;
+in
+  unstablePkgs.gitRepo.overrideAttrs(oldAttrs: rec {
+    version = "2.45";
+
+    src = fetchFromGitHub {
+      owner = "android";
+      repo = "tools_repo";
+      rev = "v${ version }";
+      hash = "sha256-f765TcOHL8wdPa9qSmGegofjCXx1tF/K5bRQnYQcYVc=";
+    };
+
+    nativeBuildInputs = (oldAttrs.nativeBuildInputs or []) ++ [ rsync  git ];
+
+    # NOTE: why `git apply` instead of relying  on `patches`? For some reason when
+    #       using `patches` the source `rsync`ed into `var/repo` is missing those changes
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p var/repo
+      rsync -a $src/ var/repo/
+
+      (
+        export GIT_CONFIG_GLOBAL=$TMPDIR/.gitconfig
+        export GIT_CONFIG_NOSYSTEM=true
+
+        cd var/repo
+
+        git config --global --add safe.directory "$PWD"
+        git config --global user.email "nemo@nix"
+        git config --global user.name "Nemo Nix"
+
+        chmod +w -R .
+
+        git init
+        git add -A
+        git commit -m "Upstream sources"
+
+        git apply ${ ./00001-add-support-for-repo2nix.patch }
+        git add -A
+        git commit -m "Patch with repo2nix"
+
+        git log -n 1 --format="%H" > ../../COMMITED_REPO_REV
+      )
```

Comment:
This is quite a hack. Would this perhaps be possible without producing a git repo?

Reply:
Unfortunately — as far as I can tell — `repo` needs it to be a git repo, as it basically clones itself out of it (see: https://gerrit.googlesource.com/git-repo/+/refs/heads/main/repo#674).

At first I tried to use `fetchFromGitHub` and `fetchFromGit` to produce a suitable git repo, but couldn't get it to work properly even with `leaveDotGit` – for example, patches didn't apply correctly (because they didn't get committed, I assume) and it _somehow_ poisoned the git branch used for manifest repo, though I didn't trace why exactly. And only then reached out for what you see, which worked.

For this PR I'd rather avoid patching `repo` further than it already was – if it needs a git repo to clone itself out of, I'd rather provide a git repo in the store at this stage, instead of figuring out how to patch it to not need git for self–bootstrap. It can always be improved in the future, if it turns out to be a problem.

There's possibly one more way to do this, as it seems that `repo` supports using a https://www.git-scm.com/docs/git-bundle to get its sources, but it doesn't really help with having to provide patched sources (either hosted, which I understand we didn't want, or massaged in a derivation, Like I've done). The only thing it would help with is that it would be a bundle in the store, not a git repo — which might be a bit cleaner conceptually, but not sure if solves any practical problem?